engine/runtime/src/main/java/org/enso/interpreter/runtime/scope/ModuleScope.java

```diff
@@ -295,7 +310,7 @@ public void reset() {
   public ModuleScope withTypes(List<String> typeNames) {
     Map<String, Object> polyglotSymbols = new HashMap<>(this.polyglotSymbols);
     Map<String, Type> requestedTypes = new HashMap<>(this.types);
-    Map<Type, Map<String, Function>> methods = new HashMap<>();
+    Map<Type, Map<String, Function>> methods = new ConcurrentHashMap<>();
     Map<Type, Map<Type, Function>> conversions = new HashMap<>();
```

Comment:
Thru-out our code base the support for `conversion` is usually direct copy of support for methods. Whatever solution we choose for `methods`, we should do the same for `conversions`.

Reply:
Not entirely, `conversions` were unlikely to have a concurrent access from what I saw so I didn't feel like there is a need to change it. But we can take the same approach, sure.